### ISSUE TITLE
Crash When Opening OptiScaler

### ISSUE THREAD

**Issue by spooknik** [emilwojcik93/RDR2-DLSS-FG-Modding-Guide#1]
Hello! Firstly, thanks for putting together this guide. Saved me so much googling. 

I have a fresh install of RDR2 and I installed DLSS-Enabler (version.dll) and then OptiScaler (v0.7.0-pre66) by renaming nvngx.dll to dlss-enabler-upscaler.dll. 

At the menu when I press `Insert` OptiScaler pops up but says I need to turn on DLSS in the graphics settings. I do that and set DLSS to Auto. But then when I press `Insert` again, it crashes the game. I also tried in-game to do it and the same. 

Any idea what could be the issue?

GPU: RTX 3070 (laptop)

**Comment by emilwojcik93** [comment]
Hi,

Please ensure you are using [Graphic API DirectX 12](https://steamcommunity.com/app/1174180/discussions/0/3762229949248826089/) and uninstall the current mod. There is a comprehensive package available under a different tag: [artur-graniszewski/DLSS-Enabler 3.03.000.0 TRUNK](https://github.com/artur-graniszewski/DLSS-Enabler/releases/tag/3.03.000.0-trunk). Installing this package will complete the steps for both DLSS-Enabler and OptiScaler.

If the issue persists, please follow these steps:

1. Remove DLSS-Enabler and all related artifacts via `uninstall.exe` in the RDR2 game root path.
2. Remove your [RDR2 settings](https://www.pcgamingwiki.com/wiki/Red_Dead_Redemption_2#Configuration_file.28s.29_location).
3. Start the game, go to in-game settings, and perform auto configuration (ensure Rockstar Launcher does not restore cloud settings by disabling cloud sync in the Launcher).
4. Enable DLSS with the preset ultra performance or performance via the in-game settings panel.
5. In the "Advanced" settings, set [Graphic API to DirectX 12](https://steamcommunity.com/app/1174180/discussions/0/3762229949248826089/) (OptiScaler works only with DX12).
6. Close the game.
7. Install [artur-graniszewski/DLSS-Enabler 3.03.000.0 TRUNK](https://github.com/artur-graniszewski/DLSS-Enabler/releases/tag/3.03.000.0-trunk).
8. Start the game again and check DLSS-Enabler and OptiScaler settings via the overlay `Insert` hotkey.

Alternatively, you can change the Graphics API by editing the `settings.xml` file:
1. Navigate to the configuration file location:
   ```
   C:\Users\<YourUsername>\Documents\Rockstar Games\Red Dead Redemption 2\Settings
   ```
2. Open the `system.xml` file with a text editor like Notepad.
3. Find the line that specifies the graphics API:
   ```xml
   <API>kSettingAPI_Vulkan</API>
   ```
4. Change `Vulkan` to `DX12`:
   ```xml
   <API>kSettingAPI_DX12</API>
   ```
5. Save the file and close the text editor.
6. Launch the game.

For more details, you can refer to this guide: [Change RDR2 Graphics API (Vulkan/DX12) using Notepad](https://www.techrbun.com/change-rdr2-graphics-api-vulkan-dx12-using-notepad/).

Please let me know what was the cause of the issue and if my guide helped somehow.

Best regards,  
emilwojcik93.

**Comment by emilwojcik93** [comment]
Hi, I just updated [README.md](https://github.com/emilwojcik93/RDR2-DLSS-FG-Modding-Guide/blob/main/README.md) with info about:
- Graphics API DX12
- enabling in-game DLSS

**Comment by spooknik** [comment]
Thanks for the detailed write up. 

The main problem was I was still on Vulkan API, so changing to DX12 probably fixed the main issue. 

Unfortunately DLSS-Enabler 3.03.000.0 did not work for me. The files installed correctly and I chose OptiScaler during the install but when pressing `insert` in game, it did not come up. Tried both `winmm.dll` and `version.dll`

But going back to the method you posted before (install DLSS-Enabler 3.02.000.0, then OptiScaler rename `dlss-enabler-upscaler.dll` works for me now.
